example/example/spiders/books_04_overrides_02.py

```diff
@@ -63,19 +62,10 @@ class BooksSpider(scrapy.Spider):
     # Configuring different page objects pages for different domains
     custom_settings = {
         "SCRAPY_POET_OVERRIDES": [
-            ("toscrape.com", BTSBookListPage, BookListPage),
-            ("toscrape.com", BTSBookPage, BookPage),
-            # We could also use the long-form version if we want to.
-            ApplyRule(
-                for_patterns=Patterns(["bookpage.com"]),
-                use=BPBookListPage,
-                instead_of=BookListPage,
-            ),
-            ApplyRule(
-                for_patterns=Patterns(["bookpage.com"]),
-                use=BPBookPage,
-                instead_of=BookPage,
-            ),
+            ApplyRule("toscrape.com", use=BTSBookListPage, instead_of=BookListPage),
+            ApplyRule("toscrape.com", use=BTSBookPage, instead_of=BookPage),
+            ApplyRule("bookpage.com", use=BPBookListPage, instead_of=BookListPage),
+            ApplyRule("bookpage.com", use=BPBookPage, instead_of=BookPage),
```

Comment:
TIL that this website exists! Good catch!

We can have this on a separate PR to reduce the cognitive workload needed on this already large PR. Created this issue for tracking: https://github.com/scrapinghub/scrapy-poet/issues/107

Reply:
I’m OK with the URL changes being made separately.

But I think we might want to remove these specific lines (and a similar case somewhere else in the changes) in this PR. I believe they used to exist to showcase how you could use either a class or a tuple to pass rules, and now that you only need a class, rather than converting the tuple ones to classes, we should remove the tuple ones, simplifying the example in the process.

Reply:
> I believe they used to exist to showcase how you could use either a class or a tuple to pass rules, and now that you can only use a class, rather than converting the tuple ones to classes,

Yes, that's right.

However, they also showcase that a single PO can be overridden by one or more POs. An example would be from `BookListPage`:
```python 
ApplyRule("toscrape.com", use=BTSBookListPage, instead_of=BookListPage)
ApplyRule("bookpage.com", use=BPBookListPage, instead_of=BookListPage)
```

From all of these example files, only `books_04_overrides_02` has this. Albeit its docstring doesn't explicitly mention it.

Reply:
I still think that we should keep them, since `books_04_overrides_02.py` is the only code example file that showcases multiple overrides for a single PO. For example:

```python
ApplyRule("toscrape.com", use=BTSBookListPage, instead_of=BookListPage)
ApplyRule("bookpage.com", use=BPBookListPage, instead_of=BookListPage)
```